Update structure and use webpack

KABBOUCHI opened the pull request:
- Update structure (more friendly)
- Add config file
- Add webpack: client will use the precompiled version
- Auth

user no longer need to add manually, install or add options

![image](https://user-images.githubusercontent.com/9212274/33847715-2b53238c-deb4-11e7-8366-2eb9141c1750.png)



N.B: Not fully tested, please check if Controller need updates 

Inspired by Laravel Horizon

KABBOUCHI committed:
Change structure and add webpack.mix.js

KABBOUCHI committed:
Change structure and add webpack.mix.js

KABBOUCHI committed:
fix defaults

KABBOUCHI committed:
fix defaults

KABBOUCHI committed:
remove invalid instructions

KABBOUCHI committed:
remove invalid instructions

ctf0 commented on README.md:
````diff
@@ -28,32 +28,6 @@ A file based translation manager, which unlike other Lang managers don't need a
-- install dependencies
-
-```bash
-yarn add vue axios vue-ls vuemit vue-notif vue-clipboard2 vue-tippy vue-awesome
````
the reason for keeping the dependencies external is so you can use the same package anywhere else without the need to reinstall it

KABBOUCHI replied:
No need to install any dependencies, only the dev need them

ctf0 commented on README.md:
```diff
@@ -63,22 +37,4 @@ new Vue({
-
-- for styling we use ***bulma***
-
-> ***Or Use another Framework***
```
not sure why ppl get confused with this, but u r not the first one to fall into this quote trap.

KABBOUCHI replied:
You can use other style:
1- .env (or config('lingo.theme')) => LINGO_THEME=theme1
2- publish lingo-views and add new file lingo-theme1.blade.php


![image](https://user-images.githubusercontent.com/9212274/33849185-8d922508-deb8-11e7-8373-e6e5d511238f.png)

ctf0 commented on config/lingo.php:
```diff
@@ -0,0 +1,8 @@
+    'as'         => 'lingo.',
+    'middleware' => 'web',
+    'theme'      => env('LINGO_THEME','bulma'),
+];
```
why do u need this ?

KABBOUCHI replied:
add ability for user to change
- the prefix (route)
- the theme from config file or .env (previously was MIX_LINGO_FRAMEWORK) 
- custom middleware

ctf0 commented on package.json:
```diff
@@ -0,0 +1,30 @@
+    "es6-promise": "^4.0.5",
+    "laravel-mix": "^1.4.2"
+  }
+}
```
again why do u need this ? other than restricting the packages versions

KABBOUCHI replied:
You always need to restrict package version,  from v1 to v2 can break things

and again this for the developer

ctf0 commented on public/css/app.css:
```diff
@@ -0,0 +1,55 @@
+}
+
+
+/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIndlYnBhY2s6Ly8vLi9yZXNvdXJjZXMvYXNzZXRzL3Nhc3MvcmVzb3VyY2VzL2Fzc2V0cy9zYXNzL2FwcC5zY3NzIiwid2VicGFjazovLy8uL2FwcC5zY3NzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBOztFQUVJO0VBQ0E7Q0NDSDs7QURFRDtFQUNJO0NDQ0g7O0FERUQ7RUFDSTtDQ0NIOztBREVEOztFQUVJO0VBQUE7Q0NDSDs7QURFRDs7RUFFSTtVQUFBO0VBQ0E7Q0NDSDs7QURFRDs7RUF ...
```
what about the `bulma` override styles ?

KABBOUCHI replied:
default will be bulma.css, you can support other framework like bootstrap

when user need to add custom style he can create new theme and change the file

![image](https://user-images.githubusercontent.com/9212274/33849483-678495ac-deb9-11e7-972e-e3cfad8f808e.png)

KABBOUCHI committed:
publish views

KABBOUCHI committed:
publish views

KABBOUCHI committed:
fix publish namings

KABBOUCHI committed:
fix publish namings

KABBOUCHI committed:
Add Authentication like Horzion

You will only be able to access lingo in the local environment.

The auth method accepts a callback which should return true or false, indicating whether the user should have access to the Lingo section

Lingo::auth(function ($request) {
    // return true / false;
});

KABBOUCHI committed:
Add Authentication like Horzion

You will only be able to access lingo in the local environment.

The auth method accepts a callback which should return true or false, indicating whether the user should have access to the Lingo section

Lingo::auth(function ($request) {
    // return true / false;
});

KABBOUCHI commented:
I added some new updates like Lingo::Auth

KABBOUCHI committed:
remove jquery

KABBOUCHI committed:
remove jquery

KABBOUCHI committed:
compile minified js/css

KABBOUCHI committed:
compile minified js/css

KABBOUCHI committed:
remove `lingo.as` option

KABBOUCHI committed:
remove `lingo.as` option

ctf0 committed:
fix laravel version

KABBOUCHI committed:
Merge branch 'master' of https://github.com/KABBOUCHI/Lingo

ctf0 commented:
first thanx so much for all ur help

- the config is not really needed, because for the routes in its current form, you are free to do whatever you want with it, you can put it under any group of middlewares, prefixes, etc..

config should be used only when its needed, specially as atm laravel config folder is a big mess.

- for the webpack, could you tell how will i use it in a fresh laravel install ?

- for changing the folder structure, i prefer to keep all the package files under the same level `src`, so plz undo that.

- all my packages are made toward devs, so maintaining a compiled `css, js` files is simply a big headache for me.

- for the controller, am not sure what exactly have you changed other than using the config values.

- am not a big fan of the horizon auth approach, so this also will be declined.

ctf0 commented:
@KABBOUCHI why closed ?

KABBOUCHI commented:
you declined all my updates, why you need it :)

edit:

all my updates does is for the end user benefits, if you don't want to maintain your product, it's okay no need to merge it

ctf0 commented:
how about this 
> for the webpack, could you tell how will i use it in a fresh laravel install ?

KABBOUCHI commented:
Not sure what do you want, I added webpack to precompile css/js for dev purpose

it's already there in a fresh laravel installer

ctf0 commented:
okay, thanx again
